### PR TITLE
add cmc provider to get quotes

### PR DESCRIPTION
**Pull request by Pantani**
add [CoinMarketCap](https://coinmarketcap.com/) provider to get quotes

### CODE REVIEW

File: marketdata/market/coinmarketcap/cmc.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	tokenId := ""
+	symbol := price.Symbol
+	coinType := blockatlas.TypeCoin
+	if price.Platform != nil {
+		coinType = blockatlas.TypeToken
+		symbol = price.Platform.Symbol
+		tokenId = price.Symbol
+	}
+
+	//[{0 BTC  coin {223.5 10 USD cmc} 2019-11-13 20:04:23.67621 -0300 -03 m=+0.003232497 } {0 ETH  coin {11.11 20 USD cmc} 2019-11-13 20:04:23.676211 -0300 -03 m=+0.003233090 }], want [{0 BTC  coin {223.55 10 USD cmc} 1969-12-31 21:05:33 -0300 -03 } {0 ETH  coin {11.11 20 USD cmc} 1969-12-31 21:05:33 -0300 -03 }]
```
Comment (dpereskokov) [inline marketdata/market/coinmarketcap/cmc.go]: lets remove this comment

File: marketdata/market/coinmarketcap/cmc.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			UpdateTime: time.Second * 30,
+		},
+	}
+	m.Headers["X-CMC_PRO_API_KEY"] = viper.GetString("market.cmc_api_key")
+	return m
+}
+
+func (m *Market) GetData() (blockatlas.Tickers, error) {
+	var prices CoinPrices
+	err := m.Get(&prices, "v1/cryptocurrency/listings/latest", url.Values{"limit": {"5000"}, "convert": {"USD"}})
```
Comment (dpereskokov) [inline marketdata/market/coinmarketcap/cmc.go]: what do you think of moving network related code to separate client.go as it's been done for blockchain classes?

File: marketdata/market/coinmarketcap/cmc.go (hunk 1 above)
Comment (Pantani) [reply marketdata/market/coinmarketcap/cmc.go]: oh, I forgot this

File: marketdata/market/coinmarketcap/cmc.go (hunk 2 above)
Comment (Pantani) [reply marketdata/market/coinmarketcap/cmc.go]: it's a lot layer for a simple thing. Make sense if the method getData have a lot of lines

File: marketdata/market/coinmarketcap/cmc.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			Request:    blockatlas.InitClient(viper.GetString("market.cmc_api")),
+			UpdateTime: time.Second * 30,
+		},
+	}
+	m.Headers["X-CMC_PRO_API_KEY"] = viper.GetString("market.cmc_api_key")
+	return m
+}
+
+func (m *Market) GetData() (blockatlas.Tickers, error) {
+	var prices CoinPrices
```
Comment (dpereskokov) [inline marketdata/market/coinmarketcap/cmc.go]: Run fmt please. Something wrong with spaces.

**Pull request merged**